src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -1093,8 +1096,11 @@ public DType ToTable(ref bool fError)
 
             switch (Kind)
             {
+                case DKind.LazyTable:
                 case DKind.Table:
                     return this;
+                case DKind.LazyRecord:
+                    return new DType(LazyTypeProvider, isTable: true);
                 case DKind.Record:
                 case DKind.DataEntity:
                 case DKind.Control:
```

Comment:
Do we still need below check which was removed by your change?

    if (ExpandInfo != null)
                    {
                        return new DType(DKind.Table, ExpandInfo, TypeTree);
                Contracts.Assert(false, "Bad source kind for ToTable");
            }

Reply:
Oh, ouch. I missed that these implementations of toTable diverged for ExpandInfo. That's not great...